osu.Game/Online/Rooms/MultiplayerBeatmapTracker.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Game.Beatmaps;
+
+namespace osu.Game.Online.Rooms
+{
+    public class MultiplayerBeatmapTracker : DownloadTrackingComposite<BeatmapSetInfo, BeatmapManager>
+    {
+        public readonly IBindable<PlaylistItem> SelectedItem = new Bindable<PlaylistItem>();
+
+        /// <summary>
+        /// The availability state of the currently selected playlist item.
+        /// </summary>
+        public IBindable<BeatmapAvailability> Availability => availability;
+
+        private readonly Bindable<BeatmapAvailability> availability = new Bindable<BeatmapAvailability>();
+
+        public MultiplayerBeatmapTracker()
+        {
+            State.BindValueChanged(_ => updateAvailability());
+            Progress.BindValueChanged(_ => updateAvailability());
+            updateAvailability();
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            SelectedItem.BindValueChanged(item => Model.Value = item.NewValue?.Beatmap.Value.BeatmapSet, true);
+        }
+
+        protected override bool VerifyDatabasedModel(BeatmapSetInfo databasedSet)
+        {
+            int? beatmapId = SelectedItem.Value.Beatmap.Value.OnlineBeatmapID;
+            string checksum = SelectedItem.Value.Beatmap.Value.MD5Hash;
+
+            BeatmapInfo matchingBeatmap;
+
+            if (databasedSet.Beatmaps == null)
```

Comment:
when does this happen?

Reply:
when deleting the beatmap, `ItemRemoved` would be invoked with a beatmapset that has `Beatmaps = null`.

Reply:
not sure how to feel about letting local bypasses like this spread but fair enough i guess

Reply:
I thought the comment inside would be enough?

Reply:
it says what's happening, not why. i don't know what a "usable state" means. mention deleting in the comment, if the scenario this is fixing is about deleting.

Reply:
fair point

Reply:
actually, I've removed the whole thing as I realize now this method will only be called after importing.

tested with the ready button using the dependent PR #11516 and it's working without any crashes.